src/plat/qemu-arm-virt/config.cmake

```
@@ -34,9 +37,32 @@ endmacro()
 
 if(KernelPlatformQEMUArmVirt)
 
+    # Our default QEMU configuration is AARCH64/Cortex-a53. If KernelSel4Arch is
+    # set to aarch32, the default is Cortex-a15. Setting ARM_CPU allows picking
+    # a specific CPU and forces KernelSel4Arch to the respective architectue.
+    # Note that the QEMU virt platform has some limitation:
+    # - only specific ARMv7 and ARMv8 cores can be used.
+    # - ARMv8 cores can't be started in AARCH32 mode
+    # - Cortex-A32 that implements AARCH32 only is not suppoted
+
     if(NOT ARM_CPU)
-        message(STATUS "ARM_CPU not set, defaulting to cortex-a53")
-        set(ARM_CPU "cortex-a53")
+        # We don't have any KernelSel4Archxxx varable set up here, because
+        # declare_seL4_arch() has no been called yet. So we must check
+        # KernelSel4Arch for a hint what to default to.
+        if(NOT KernelSel4Arch)
+            set(KernelSel4Arch "aarch64")
+        endif()
+        set(
+            arch_cpu_mapping # element format: "KernelSel4Arch:ARM_CPU"
+            "aarch64:cortex-a53"
+            "arm_hyp:cortex-a15"
+            "aarch32:cortex-a15"
+        )
+        if(NOT ";${arch_cpu_mapping};" MATCHES ";${KernelSel4Arch}(:([^;]*))?;")
```

The `(:..)?` part looks strange to me. Why do we allow that to be optional? Don't we want to force a match?

i.e.
```suggestion
        if(NOT ";${arch_cpu_mapping};" MATCHES ";${KernelSel4Arch}:([^;]*);")
```

(This might change the match number, not sure, i.e. potentially `CMAKE_MATCH_1` below)

This will only really be relevant if we mess up the format in the table above, but I'd still prefer the version that forces the second part to exist.